waiter/src/waiter/util/client_tools.clj

```diff
@@ -1196,11 +1200,11 @@
 (defn post-token
   "Sends a POST request with the given token definition"
   [waiter-url {:keys [token] :as token-map} &
-   {:keys [cookies headers query-params] :or {cookies [] headers {} query-params {}}}]
+   {:keys [cookies headers query-params use-host-header] :or {cookies [] headers {} query-params {} use-host-header true}}]
```

Comment:
Instead of the `use-host-header` flag, can we not require the client to pass in the headers? I think Waiter will prefer `x-waiter-token` header over host header.